src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp

```diff
@@ -75,6 +79,24 @@ bool GetEntrypointExecutableAbsolutePath(std::string& entrypointExecutable)
             result = true;
         }
     }
+#elif defined (__FreeBSD__)
+    static const int name[] = {
+        CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1
+    };
+    char path[PATH_MAX];
+    size_t len;
+
+    len = sizeof(path);
+    if (sysctl(name, 4, path, &len, NULL, 0) == 0)
```

Comment:
That was my first though as well, but it turns out `__arraycount()` is actually not by defined for FreeBSD unless you include `/usr/src/lib/libnetbsd/cdefs.h`. Which would mean we would need to probe for more stuff in the CMAKE-section which strictly speaking is not needed here. So I thought it was better without it. Any objections?